### PR TITLE
clown car can no longer be cheesed

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
-clown car sucks people up if it runs them over whilst they are laying down
-clown car occupants are restrained, so they cant shoot the car from inside

## Why It's Good For The Game
spending 20 tc for something people can beat just by pressing the r"rest" button everyone gets is a real shame

## Changelog
:cl:
balance: clown car occupants are now restrained
balance: clown car can no longer be cheesed by laying down
/:cl:

### CODE REVIEW

**Comment by TotallyNotChief** [comment]
cant you click on people to manually scoop them up bruh

**Comment by zeskorion** [comment]
> cant you click on people to manually scoop them up bruh

it takes a second to do so. I made this when i saw a clown car get bolted in a room with a tider with a baseball bat, who just laid down and moved about, being totally immune to numerous attempts by the clown to scoop them up.

**Comment by PowerfulBacon** [comment]
How are you meant to stop it now?

**Comment by zeskorion** [comment]
> How are you meant to stop it now?

shooting it, AI bolting doors, outmaneuvering it, etc.

**Comment by Mick1299** [comment]
Can you still stand on a table to avoid it?

**Comment by Archanial** [comment]
Laying down isn't a cheese, it's a downside or a counter.

**Comment by Tamuren** [comment]
Don't merge this, clown car is effective when it surprises. CC moves quickly anyway so this rebalance would be OP.

**Comment by WhyIsCaeciliusTaken** [comment]
Literally just yell ;AI law 2 disable grav gen and the car is stopped

**Comment by zeskorion** [comment]
> Can you still stand on a table to avoid it?

Yes. It cant pass through tables and this strat does not allow trivializing it like resting does, and therefore will remain unchanged



> Laying down isn't a cheese, it's a downside or a counter.

It's a 20tc item that can be trivialized by an assistant with a baseball bat

**Comment by Mick1299** [comment]
Good change, speedmerge

**Comment by TotallyNotChief** [comment]
>It's a 20tc item that can be trivialized by an assistant with a baseball bat

local coders fails at clicking spacemen, decides to make the car uncounterable

**Comment by zeskorion** [comment]
> > It's a 20tc item that can be trivialized by an assistant with a baseball bat
> 
> local coders fails at clicking spacemen, decides to make the car uncounterable 

uncounterable? just robust it the old fashioned way, with a good old pair of bolted doors, or even just a fastened-down locker, table, crate, etc...

File: code/modules/mob/living/carbon/human/human.dm [hunk 1]
```diff
@@ -207,8 +207,11 @@
 // this could be made more general, but for now just handle mulebot
 /mob/living/carbon/human/Crossed(atom/movable/AM)
 	var/mob/living/simple_animal/bot/mulebot/MB = AM
+	var/obj/vehicle/sealed/car/C = AM
 	if(istype(MB))
 		MB.RunOver(src)
+	if(istype(C))
```
Comment (ike709) [inline code/modules/mob/living/carbon/human/human.dm]: Is there not a `Crossed()` component signal?

At least make this an else if.

Comment (zeskorion) [reply code/modules/mob/living/carbon/human/human.dm]: Done. I made this pr about two and a half weeks before i finally figured out components

**Comment by ike709** [comment]
Can't wait to see how this is abused.

**Pull request merged**